core/src/main/java/org/elasticsearch/indices/IndicesModule.java

```diff
@@ -138,6 +237,15 @@ protected void configure() {
         bind(NodeServicesProvider.class).asEagerSingleton();
     }
 
+    // for testing
+    public synchronized MapperRegistry getMapperRegistry() {
```

Comment:
I don't think it would be easy. In general several tests (from different packages) need a way to easily register the base mappers so that they can deserialize arbitrary mappings, so there is a need for something that registers the main mappers. An alternative would be to duplicate the logic of IndicesModules.register* in the test-framework.